Mathlib/Data/Polynomial/Laurent.lean

```diff
@@ -610,4 +625,32 @@ theorem isLocalization : IsLocalization (Submonoid.closure ({X} : Set R[X])) R[T
 
 end CommSemiring
 
+section Inversion
+
+variable {R : Type*} [CommSemiring R]
+
+/-- The map which substitutes `T ↦ T⁻¹` into a Laurent polynomial. -/
+def invert : R[T;T⁻¹] ≃ₐ[R] R[T;T⁻¹] := AddMonoidAlgebra.domCongr R R <| AddEquiv.neg _
+
+@[simp] lemma invert_T {n : ℤ} : invert (T n : R[T;T⁻¹]) = T (-n) :=
+  AddMonoidAlgebra.domCongr_single _ _ _ _ _
+
+@[simp] lemma invert_apply {f : R[T;T⁻¹]} {n : ℤ} : invert f n = f (-n) := rfl
+
+@[simp] lemma invert_comp_C : invert ∘ (@C R _) = C := by ext; simp
+
+@[simp] lemma invert_C {t : R} : invert (C t) = C t := by ext; simp
+
+lemma involutive_invert : Involutive (invert (R := R)) := fun _ ↦ by ext; simp
```

Comment:
Can you add the related lemma that `(invert (R := R)).symm = invert` (which I think is `rfl`?)

Reply:
Done!